mlflow/store/sqlalchemy_store.py

```diff
@@ -206,7 +205,7 @@ def create_run(self, experiment_id, user_id, run_name, source_type, source_name,
         run_uuid = uuid.uuid4().hex
         artifact_location = build_path(experiment.artifact_location, run_uuid,
                                        SqlAlchemyStore.ARTIFACTS_FOLDER_NAME)
-        run = SqlRun(name=run_name or "", artifact_uri=artifact_location, run_uuid=run_uuid,
+        run = SqlRun(name=run_name, artifact_uri=artifact_location, run_uuid=run_uuid,
```

Comment:
It would be great if the store implementations (or at least sqlalchemy) could go ahead and read from the tag and then the run_name parameter if not available. This way, they'll be forwards-compatible when we stop sending run_name via RestStore.

Reply:
In the current PR I've tried to make the logic around run name consistent with other fields, i.e. the logic for determining attributes from their corresponding tags takes place in the `MlflowClient`. The stores then don't have to reimplement this logic - they just pass through the attributes and tags they're passed to their respective backends.

I think that determining the attributes from tags in the stores would make sense if we modify the store APIs to not accept run_name as an argument before they've been removed from their backends / the protobuf files. However, before that happens I think centralising the logic in `MlflowClient` is best.